src/ui-lib/sass/05-components/01-atoms/06-buttons/_08-toggle-menu.scss

```diff
@@ -1,31 +1,105 @@
+$grav-sp-icon: 0.35rem;
+$grav-tr-icon-hover: 0.1rem;
+$grav-tr-icon-rotate: 135deg;
+$grav-tr-icon-scale: 1.2;
+$grav-size-icon: 33px;
+
 .grav-c-toggle-menu {
+  display: flex;
+  height: $grav-size-icon;
   margin-top: 0;
-  border: $grav-st-thickness solid $grav-co-neutral-asphalt;
+  border: hidden;
   background-color: inherit;
-  color: $grav-co-bg;
-
-  &:hover {
+  text-align: center;
+  
+  &--inverted,
+  .is-light & {
+    &__icon,
+    &__icon::before,
+    &__icon::after {
+      background-color: $grav-co-neutral-asphalt;
+    }
+     
+    &[aria-pressed='true']:hover,
+    &[aria-pressed='true']:active,
+    &[aria-pressed='true']:focus {
+      background-color: inherit;
+    }
+  }
+  
+  &:hover,
+  &:active,
+  &:focus {
     background-color: inherit;
     cursor: pointer;
+
   }
 
-  &:active,
-  &:focus {
-    color: $grav-co-bg;
+  &__icon {
+    display: inline-block;
+    width: $grav-sp-m;
+    height: $grav-st-thickness;
+    position: relative;
+    margin-top: auto;
+    background-color: $grav-co-bg;
+    transition: all $grav-tr-speed-fast;
+
+    &::before,
+    &::after {
+      content: '';
+      display: inline-block;
+
+      width: $grav-sp-m;
+      height: $grav-st-thickness;
+      position: absolute;
+      left: 0;
+      background-color: $grav-co-bg;
+
+      transition: all $grav-tr-speed-fast;
+    }
+
+    &::before {top: -$grav-sp-icon;}
+    
+    &::after {top: $grav-sp-icon;}
+
+  }
+  
+  &:hover &__icon::before {
+    transform: translateY(-$grav-tr-icon-hover);
+  }
+
+  &:hover &__icon::after {
+    transform: translateY($grav-tr-icon-hover);
   }
 
   &[aria-pressed='true'] {
     background-color: inherit;
-    color: $grav-co-bg;
+    transition: all $grav-tr-speed-fast;
```

Comment:
Don't need the `transition` property on the button itself, since nothing changes. You're only doing transitions on the icon elements within it.